### PR TITLE
[Merged by Bors] - feat: port SetTheory.Game.Basic

### PR DESCRIPTION
**Pull request by GabinKolly**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by urkud** [comment]
There was no activity here for ~2 weeks, so I assume that you are not actively working on this PR. I'll add `help-wanted` so that anyone can contribute without an explicit permission from the author.

**Comment by dwrensha** [comment]
I got `quot_left_distrib` and `quot_mul_assoc` working. Now the only remaining task I see is to resolve some `simpNF` linter issues.

**Comment by semorrison** [comment]
@Parcly-Taxel the new lemmas you should added, e.g.
```
@[simp]
theorem neg_mk_mul_moveLeft_inl' {xl xr yl yr} {xL xR yL yR} {i j} :
    -(mk xl xr xL xR * mk yl yr yL yR).moveRight
    (@FunLike.coe _ _ (fun _ => RightMoves (mk xl xr xL xR * mk yl yr yL yR))
      Equiv.instFunLikeEquiv toLeftMovesNeg.symm (Sum.inl (i, j))) =
    -(xL i * mk yl yr yL yR + mk xl xr xL xR * yR j - xL i * yR j) :=
  rfl
```
seem unlikely to be useful to me. At least for this file, we don't need them. Could you either:
* remove them
* working out how to reformulate them in some way that is still useful, but doesn't have `@FunLike.coe`?

**Comment by Parcly-Taxel** [comment]
> @Parcly-Taxel the new lemmas you just added, e.g.
> 
> ```
> @[simp]
> theorem neg_mk_mul_moveLeft_inl' {xl xr yl yr} {xL xR yL yR} {i j} :
>     -(mk xl xr xL xR * mk yl yr yL yR).moveRight
>     (@FunLike.coe _ _ (fun _ => RightMoves (mk xl xr xL xR * mk yl yr yL yR))
>       Equiv.instFunLikeEquiv toLeftMovesNeg.symm (Sum.inl (i, j))) =
>     -(xL i * mk yl yr yL yR + mk xl xr xL xR * yR j - xL i * yR j) :=
>   rfl
> ```
> 
> seem unlikely to be useful to me. At least for this file, we don't need them. Could you either:
> 
>     * remove them
> 
>     * working out how to reformulate them in some way that is still useful, but doesn't have `@FunLike.coe`?

I don't know how to get the `FunLike.coe` out –
```
toRightMovesNeg.symm (Sum.inr (i, j))
```
as shown in the infoview doesn't work.

File: Mathlib/SetTheory/Game/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,962 @@
...
+@[simp]
+theorem mul_moveRight_inr {x y : PGame} {i j} :
+    (x * y).moveRight (toRightMovesMul (Sum.inr (i, j))) =
+      x.moveRight i * y + x * y.moveLeft j - x.moveRight i * y.moveLeft j := by
+  cases x
+  cases y
+  rfl
+#align pgame.mul_move_right_inr PGame.mul_moveRight_inr
+
+-- @[simp] -- Porting note: removed
```
Comment (semorrison) [inline Mathlib/SetTheory/Game/Basic.lean:343]: should say why, please: "simpNF linter complains".

**Comment by semorrison** [comment]
Okay, I've created a [lean4 issue](https://github.com/leanprover/lean4/issues/2288) for the `termination_by` problem (or at least something closely related?).

@Parcly-Taxel, if you could just fix the porting notes then I think we can merge.

File: Mathlib/SetTheory/Game/Basic.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/SetTheory/Game/Basic.lean:343]: (and below)

**Comment by Parcly-Taxel** [comment]
@semorrison OK, done.

**Comment by semorrison** [comment]
Bors merge

**Pull request closed without merging**